third_party/move/move-core/types/src/account_address.rs

```diff
@@ -336,8 +350,7 @@ impl fmt::Display for AccountAddressParseError {
     fn fmt(&self, f: &mut fmt::Formatter) -> std::fmt::Result {
         write!(
             f,
-            "Unable to parse AccountAddress (must be hex string of length {})",
-            AccountAddress::LENGTH
+            "Unable to parse AccountAddress (must be hex string that conforms to AIP-40)",
```

Comment:
I don't think this is a very use friendly message tbh.  This could be provided to an end retail user when passed through several layers.  They should probably not know or care about AIP-40.

Can we perhaps explain simply what might need to be fixed?

Reply:
Yeah fair call. I would actually prefer to include a URL to the relevant section of the AIP, what do you think about that?

Reply:
I'd probably just suggest we state it is an invalid address and maybe print out the hex version of what we saw... Otherwise you'll need more explicit parsing errors that enumerate the various different cases.